src/containers/load-collection.tsx

```diff
@@ -16,97 +17,118 @@ export interface CollectionResult {
 }
 
 interface Props extends ArsOptionsWithDeprecations {
+  sort: keyof Record
   search: string
   render: (result: CollectionResult) => React.ReactNode | null
-  page: number
 }
 
 interface State {
   data: Record[]
   error: string | null
   fetching: boolean
-  targetURL: string
+  targetUrl: string
+  search: string
+  sort: keyof Record
+  page: number
 }
 
 type Request = {
   error: string | null
   fetching: boolean
   data: Record[]
   xhr?: XMLHttpRequest
+  valid: boolean
+}
+
+function nextPage(nextProps: Props, lastState: State) {
+  if (
+    nextProps.search !== lastState.search ||
+    nextProps.sort !== lastState.sort
+  ) {
+    return 0
+  }
+
+  return lastState.page
 }
 
 class CollectionFetcher extends React.Component<Props, State> {
   static defaultProps: Props = {
     ...DEFAULT_OPTIONS,
+    sort: 'id',
     search: '',
-    render: (result: CollectionResult) => null,
-    page: 0
+    render: result => null
   }
 
-  static getDerivedStateFromProps(props: Props, lastState: State) {
-    let { listUrl, listQuery, url, page, search } = props
+  static getDerivedStateFromProps(nextProps: Props, lastState: State) {
+    let { listUrl, listQuery, url, sort, search } = nextProps
 
+    let page = nextPage(nextProps, lastState)
     let baseUrl = listUrl(url)
-    let query = { search, page }
+    let query = { search, page, sort }
     let queryString = stringify(listQuery(query))
 
-    if ('makeURL' in props) {
-      baseUrl = props.makeURL(url)
+    if ('makeURL' in nextProps) {
+      baseUrl = nextProps.makeURL(url)
 
       console.warn(
         'ArsArsenal option makeURL is deprecated. Use listUrl instead.'
       )
     }
 
-    if ('makeQuery' in props) {
-      queryString = props.makeQuery(search)
+    if ('makeQuery' in nextProps) {
+      queryString = nextProps.makeQuery(search)
 
       console.warn(
         'ArsArsenal option makeQuery is deprecated. Use buildQuery instead.'
       )
     }
 
-    return {
-      targetURL: baseUrl + '?' + queryString
-    }
+    return { page, sort, search, targetUrl: baseUrl + '?' + queryString }
```

Comment:
This is kind of a gross function. Hopefully it can get simpler when we drop the deprecations. I need to do all of this work in here to avoid a race condition with `shouldComponentUpdate` in other modules. Doesn't feel great.